lib/model/store.dart

```diff
@@ -345,6 +345,64 @@ abstract class GlobalStore extends ChangeNotifier {
     ));
   }
 
+  /// Attempt to refresh the realm metadata for each account.
+  ///
+  /// Fetches server settings for each account and then calls [updateRealmData]
+  /// to update the realm metadata (name and icon).
+  ///
+  /// The updation for some accounts may be skipped if:
+  ///  - The [PerAccountStore] for the account is already loaded or is loading.
+  ///  - The version of corresponding realm server is unsupported.
+  ///
+  /// Returns immediately; the fetches and updates are done asynchronously.
+  void refreshRealmMetadata() {
+    for (final accountId in accountIds) {
+      // Avoid updating realm metadata if per account store has already
+      // loaded or is being loaded. It should be updated with data from
+      // the initial snapshot or realm update events.
+      if (_perAccountStoresLoading.containsKey(accountId)) continue;
+      if (_perAccountStores.containsKey(accountId)) continue;
+
+      final account = getAccount(accountId);
+      if (account == null) continue;
+
+      // Fetch the server settings and update the realm data without awaiting.
+      // This allows fetching server settings of all the accounts parallelly.
+      unawaited(() async {
+        final GetServerSettingsResult serverSettings;
+        final connection = apiConnection(
+          realmUrl: account.realmUrl,
+          zulipFeatureLevel: null);
+        try {
+          serverSettings = await getServerSettings(connection);
+          final zulipVersionData = ZulipVersionData.fromServerSettings(serverSettings);
+          if (zulipVersionData.isUnsupported) {
+            throw ServerVersionUnsupportedException(zulipVersionData);
+          }
+        } on MalformedServerResponseException catch (e) {
+          final zulipVersionData = ZulipVersionData.fromMalformedServerResponseException(e);
+          if (zulipVersionData != null && zulipVersionData.isUnsupported) {
+            throw ServerVersionUnsupportedException(zulipVersionData);
+          }
+          rethrow;
+        } finally {
+          connection.close();
+        }
```

Comment:
This logic is rather subtle, and it's duplicated with the other place we call `getServerSettings`, in the login flow. Can we deduplicate? Something like this:
```suggestion
        final serverSettings = await fetchServerSettings(account.realmUrl);
```

We'd arrange that as an NFC commit factoring out `GlobalStore.fetchServerSettings` (and using it in the existing place), then the commit adding this `refreshRealmMetadata` method using it.